pilot/pkg/networking/plugin/authz/authorization.go

```diff
@@ -50,6 +50,7 @@ func NewBuilderSkipIdentity(actionType ActionType, push *model.PushContext, prox
 		IsCustomBuilder: actionType == Custom,
 		Logger:          &builder.AuthzLogger{},
 		SkippedIdentity: skipped,
+		IsAmbient:       proxy.IsAmbient(),
```

Comment:
is it for all ambient? or only ztunnel?

Reply:
Only the waypoint adds the authz as an http filter which is the case covered by this PR.

Reply:
So although flag is true for ztunnel the generators are not executed because for Ztunnel it's listener filter.